pkg/components/references.go

```diff
@@ -45,7 +45,9 @@ func GetReference(c component, registry, imagePath, imagePrefix string, is *oper
 		case ComponentElasticsearchOperator:
 			registry = ECKRegistry
 		case ComponentOperatorInit,
-			ComponentCSRInitContainer:
+			ComponentCSRInitContainer,
+			ComponentPrometheus,
```

Comment:
While I think this is the correct behavior, I think you should make a separate case and add a PrometheusRegistry that these 2 new Components are used to select.